### DISCUSSION TITLE
Why clicking the full link to jump causes page to refresh

### DISCUSSION THREAD

**Discussion by flyerH** [vikejs/vike#1782] [Discussion]
this is example: https://github.com/flyerH/vike-vue-example
if link has base url, the page will full refresh
![6e9fdc4a67122e48b88d90dc7cc5bea8](https://github.com/user-attachments/assets/ca406db1-20d4-4ed6-ab8d-15aab541aa96)

if link is only the path, the page will partially updated
![5fc7027caa89c0e85b2080b47f7c08d9](https://github.com/user-attachments/assets/ce7f1d63-372e-4add-a653-7d68b80d35d7)

**Comment by brillout** [comment]
Indeed, Vike assumes any link starting with `http` to be external. Is there a reason you want to have links include the URL origin? That's quite unusual.

**Reply by flyerH** [reply]
For example, if the user input a hyperlink, it may be the same site or an external site. In this case, I need to judge whether it is the same site and change it to the path. If it is not the same site, keep the URL origin.

**Reply by brillout** [reply]
I see. PR welcome if that's something you'd be up to, should be relatively easy - see `vike/client/client-routing-runtime/skipLink.ts`.

**Reply by flyerH** [reply]
I think it’s better to keep the status quo, because the same URL origin may not all use vike, which means they may not all support client routing, which may 404 being returned.

**Reply by flyerH** [reply]
Maybe we can refer to how other frameworks handle it.

**Reply by brillout** [reply]
Correct me if I'm wrong, but it seems like on the client-side it's all predictable? In other words: `https://example.com/some-base/some-path` is equivalent to `/some-path` and we can deterministically derive between the two. That said, indeed, Vike needs to assume that absolute URLs contain the Base URL.

**Reply by brillout** [reply]
> Maybe we can refer to how other frameworks handle it.

:+1: I'd be curious to know what they do. Let me know if you find something interesting.
